relations/src/relations/update_account.rs

```diff
@@ -0,0 +1,95 @@
+use halo2_base::{
+    gates::{GateChip, GateInstructions},
+    poseidon::hasher::{spec::OptimizedPoseidonSpec, PoseidonHasher},
+    utils::BigPrimeField,
+    AssignedValue,
+};
+#[allow(unused_imports)]
+use halo2_base::{
+    Context,
+    QuantumCell::{Constant, Existing, Witness},
+};
+
+use crate::{
+    account::CircuitAccount,
+    poseidon_consts::{RATE, R_F, R_P, T_WIDTH},
+};
+
+pub struct UpdateAccountInput<F, A>
+where
+    F: BigPrimeField,
+    A: CircuitAccount<F>,
+{
+    //public inputs
+    pub old_account_hash: AssignedValue<F>,
+    pub new_account_hash: AssignedValue<F>,
+    pub operation: A::Op,
+
+    //witnesses
+    pub old_account: A,
```

Comment:
don't we need the new account as well? to at least ensure that we don't put trash to the note tree

Reply:
There is a contstraint that `new_account = update(old_account, operation)` and then hash of this is included in `note_new` which makes is impossible to put garbage there.